op-challenger/game/fault/trace/split/provider_test.go

```diff
@@ -0,0 +1,148 @@
+package split
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	mockGetError   = errors.New("mock get error")
+	mockOutput     = common.HexToHash("0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
+	mockCommitment = common.HexToHash("0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb")
+)
+
+func TestGet(t *testing.T) {
+	t.Run("ErrorBubblesUp", func(t *testing.T) {
+		mockOutputProvider := mockTraceProvider{getError: mockGetError}
+		splitProvider := SplitTraceProvider{
+			logger:     testlog.Logger(t, log.LvlInfo),
+			providers:  []types.TraceProvider{&mockOutputProvider},
+			depthTiers: []uint64{40, 20},
```

Comment:
I'm a little confused at having one trace provider but two depth tiers.  Though I suspect this becomes simpler if we just hard code that there's always two providers.